<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>836</width>
    <height>483</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_7" stretch="1,0">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="handleWidth">
       <number>4</number>
      </property>
      <widget class="QTabWidget" name="sortTab">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="tabPosition">
        <enum>QTabWidget::West</enum>
       </property>
       <property name="tabShape">
        <enum>QTabWidget::Triangular</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="filter">
        <attribute name="title">
         <string>Filter</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QToolBox" name="toolBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="currentIndex">
            <number>2</number>
           </property>
           <widget class="QWidget" name="tagsFilter">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>282</width>
              <height>252</height>
             </rect>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <attribute name="label">
             <string>Tags</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <layout class="QFormLayout" name="formLayout">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Title</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="titleFilter">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Artist</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="artistFilter">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QSpinBox" name="minGap">
                 <property name="maximum">
                  <number>1000000</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;≤ GAP [ms] ≤&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="maxGap">
                 <property name="maximum">
                  <number>1000000</number>
                 </property>
                 <property name="value">
                  <number>1000000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QSpinBox" name="minBPM">
                 <property name="maximum">
                  <number>1000</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;center&gt;&lt;p&gt;≤ BPM &amp;le;&lt;/p&gt;&lt;/center&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="maxBPM">
                 <property name="maximum">
                  <number>1000</number>
                 </property>
                 <property name="value">
                  <number>1000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QCheckBox" name="hasVideo">
               <property name="text">
                <string>Has Video</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="hasBackground">
               <property name="text">
                <string>Has Background</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="validatorFilter">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>258</width>
              <height>286</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Validator</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <widget class="QCheckBox" name="missingSongtextFile">
               <property name="text">
                <string>Missing Songtext</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="missingMandatoryTags">
               <property name="text">
                <string>Missing Mandatory Tags</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="hasUnknownTags">
               <property name="text">
                <string>Unknown Tags</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="missingMP3File">
               <property name="text">
                <string>Missing MP3</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="missingCoverFile">
               <property name="text">
                <string>Missing Cover</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="missingVideoFile">
               <property name="text">
                <string>Missing Video</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="missingBackgroundFile">
               <property name="text">
                <string>Missing Background</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="startsAt0">
               <property name="text">
                <string>Song does not start at 0</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="wellFormed">
               <property name="text">
                <string>Is Well Formed</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="isValid">
               <property name="text">
                <string>Is Valid</string>
               </property>
               <property name="tristate">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="propertiesFilter">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>281</width>
              <height>252</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Properties</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QSpinBox" name="minLen">
                 <property name="maximum">
                  <number>1000</number>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;≤ (MP3) Length [s] ≤&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="maxLen">
                 <property name="maximum">
                  <number>1000</number>
                 </property>
                 <property name="value">
                  <number>1000</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tab_2">
        <attribute name="title">
         <string>Sort &amp;&amp; Group</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QGroupBox" name="sortList">
           <property name="title">
            <string>Sort by</string>
           </property>
           <property name="flat">
            <bool>true</bool>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QCheckBox" name="reverseSort">
              <property name="text">
               <string>Descending</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="sortTitle">
              <property name="text">
               <string>Title</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="sortArtist">
              <property name="text">
               <string>Artist</string>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupList">
           <property name="title">
            <string>&amp;Group by</string>
           </property>
           <property name="flat">
            <bool>true</bool>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_8">
            <item>
             <widget class="QComboBox" name="groupBy"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QScrollArea" name="songScroller">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string notr="true">background-color: palette(light)</string>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="songList">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>387</width>
          <height>384</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </widget>
      <widget class="SongInfo" name="songDetails" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="AudioPlayer" name="musicPlayer" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>836</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFIle">
    <property name="title">
     <string>Fi&amp;le</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Setti&amp;ngs</string>
    </property>
    <addaction name="actionSources"/>
    <addaction name="actionMidi_Output"/>
   </widget>
   <addaction name="menuFIle"/>
   <addaction name="menuSettings"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSources">
   <property name="icon">
    <iconset theme="folder">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Songdirs ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionMidi_Output">
   <property name="text">
    <string>&amp;Midi Output ...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SongInfo</class>
   <extends>QWidget</extends>
   <header>songinfo.h</header>
   <container>1</container>
   <slots>
    <signal>seek(quint64)</signal>
    <signal>play()</signal>
    <signal>pause()</signal>
    <slot>selectionChanged()</slot>
    <slot>seekTo(quint64)</slot>
    <slot>startPlayback()</slot>
    <slot>pausePlayback()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>AudioPlayer</class>
   <extends>QWidget</extends>
   <header>audioplayer.h</header>
   <container>1</container>
   <slots>
    <signal>seeking(quint64)</signal>
    <signal>started()</signal>
    <signal>paused()</signal>
    <slot>seek(quint64)</slot>
    <slot>play()</slot>
    <slot>pause()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingSongtextFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingMandatoryTags</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>110</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasUnknownTags</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingMP3File</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>162</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingCoverFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingVideoFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>missingBackgroundFile</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>240</y>
    </hint>
    <hint type="destinationlabel">
     <x>342</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startsAt0</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>266</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>341</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wellFormed</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>isValid</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>341</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>titleFilter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>325</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>artistFilter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>124</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minGap</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>164</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxGap</sender>
   <signal>valueChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>208</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>minBPM</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>140</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>255</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>maxBPM</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasVideo</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hasBackground</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>filterList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>328</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sortTitle</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>resortList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>70</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sortArtist</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>resortList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>223</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>reverseSort</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>resortList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>326</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groupList</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>regroupList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>255</y>
    </hint>
    <hint type="destinationlabel">
     <x>324</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sortList</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>resortList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>groupBy</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>regroupList()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>338</y>
    </hint>
    <hint type="destinationlabel">
     <x>8</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refreshList()</slot>
  <slot>resortList()</slot>
  <slot>filterList()</slot>
  <slot>regroupList()</slot>
 </slots>
</ui>
